lib/screen/home/home_page.dart

```diff
@@ -184,47 +224,46 @@ class HomePageState extends State<HomePage>
     }
   }
 
+  List<CompactedAssetToken> _updateTokens(List<CompactedAssetToken> tokens) {
+    tokens = tokens.filterAssetToken();
+    tokens.sortToken();
+    return tokens;
+  }
+
   @override
   Widget build(BuildContext context) {
+    super.build(context);
     final theme = Theme.of(context);
     final contentWidget =
         BlocConsumer<NftCollectionBloc, NftCollectionBlocState>(
-            buildWhen: (previousState, currentState) {
-      final diffLength =
-          currentState.tokens.length - previousState.tokens.length;
-      if (diffLength > 0) {
-        _metricClient.addEvent(MixpanelEvent.addNFT, data: {
-          'number': diffLength,
-        });
-      }
-      return true;
-    }, builder: (context, state) {
-      final hiddenTokens =
-          injector<ConfigurationService>().getTempStorageHiddenTokenIDs();
-      final sentArtworks =
-          injector<ConfigurationService>().getRecentlySentToken();
-      final expiredTime = DateTime.now().subtract(SENT_ARTWORK_HIDE_TIME);
-      return NftCollectionGrid(
-        state: state.state,
-        tokens: state.tokens
-            .where((element) =>
-                !hiddenTokens.contains(element.id) &&
-                !sentArtworks.any((e) => e.isHidden(
-                    tokenID: element.id,
-                    address: element.ownerAddress,
-                    timestamp: expiredTime)))
-            .toList(),
-        loadingIndicatorBuilder: _loadingView,
-        emptyGalleryViewBuilder: _emptyGallery,
-        customGalleryViewBuilder: (context, tokens) =>
-            _assetsWidget(context, tokens),
-      );
-    }, listener: (context, state) async {
-      log.info("[NftCollectionBloc] State update $state");
-      if (state.state == NftLoadingState.done) {
-        _onTokensUpdate(state.tokens);
-      }
-    });
+      bloc: nftBloc,
+      buildWhen: (previousState, currentState) {
+        final diffLength =
+            currentState.tokens.length - previousState.tokens.length;
+        if (diffLength > 0) {
+          _metricClient.addEvent(MixpanelEvent.addNFT, data: {
+            'number': diffLength,
+          });
+        }
+        return true;
+      },
+      builder: (context, state) {
+        return NftCollectionGrid(
+          state: state.state,
+          tokens: _updateTokens(state.tokens),
```

Comment:
Can we return a sorted list of tokens so that we won't need to sorting the list again everytime the token get updates.

Moreover, to reduce complexity when add/remove item and re-sort the list. I suggest to create a SortedList class to use. Please try with the example below

```
class SortedList<T extends Comparable<T>> {
  List<T> _list;

  SortedList() {
    _list = <T>[];
  }

  void insert(T item) {
    int i = 0;
    while (i < _list.length && item.compareTo(_list[i]) > 0) {
      i++;
    }
    _list.insert(i, item);
  }

  void addAll(List<T> items) {
    for (T item in items) {
      insert(item);
    }
  }

  void remove(T item) {
    _list.remove(item);
  }

  List<T> get items => List.unmodifiable(_list);
}
```